### PR TITLE
Added hw6

### PR DESCRIPTION
**Pull request by ocherenkov**

### CODE REVIEW

File: hw6/main.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    std::cout << "Enter number of last element of arithmetic progression: ";
+    std::cin >> last;
+
+    if (first < 1) {
+        std::cout << "First element of arithmetic progression must be more or equal 1" << std::endl;
+    } else if (step < 1) {
+        std::cout << "Step of arithmetic progression must be more or equal 1" << std::endl;
+    } else if (last < first) {
+        std::cout << "Number of last element of arithmetic progression must be more than first element" << std::endl;
+    } else {
```
Comment (helen-black) [inline hw6/main.cpp:19]: wrong conditions. these parameters are valid:
-10, -10, 5
so you should print:
-10, -20, -30, -40, -50

only one valid check is the last input > 0

File: hw6/main.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    int num;
+    std::cout << "Enter a positive number: ";
+    std::cin >> num;
+
+    if (num < 0) {
+        std::cout << "Incorrect number entered" << std::endl;
+    } else {
+        long long factorial = 1;
+        std::cout << "Factorial of number " << num << ": ";
+        for (int i = 1; i <= num; ++i) {
```
Comment (helen-black) [inline hw6/main.cpp:61]: you can start this loop from 2

File: hw6/main.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    for (int i = 1; i <= size; ++i) {
+        for (int j = 1; j < i; ++j) {
+            std::cout << " ";
+        }
+        for (int k = 1; k <= size; ++k) {
+            std::cout << "* ";
+        }
+        std::cout << std::endl;
+    }
+
```
Comment (helen-black) [inline hw6/main.cpp:104]: it is 4.e (the same code as the next)

File: hw6/main.cpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+            std::cout << "* ";
+
+        }
+        std::cout << std::endl;
+    }
+
+    std::cout << "e:" << std::endl;
+    for (int i = 1; i <= size; ++i) {
+        for (int j = 1; j <= i; ++j) {
+            std::cout << ((i + j) % 2 == 0 ? "1" : "0") << " ";
```
Comment (helen-black) [inline hw6/main.cpp:120]: you can simplify it:
` std::cout << ((i + j) % 2 == 0);`

**Review by helen-black — CHANGES_REQUESTED** [review]

File: hw6/main.cpp (hunk 3 above)
Comment (ocherenkov) [reply hw6/main.cpp:104]: Output is correct
Enter the shape size: 4
a:
****
***
**
*
b:
*
**
***
****
с:
****
 ****
  ****
   ****
d:
   ****
  ****
 ****
****
e:
1
01
101
0101

**Review by helen-black — APPROVED** [review]

**Pull request merged**